virtcontainers/container.go

```diff
@@ -510,11 +510,14 @@ func (c *Container) mountSharedDirMounts(hostSharedDir, guestSharedDir string) (
 func (c *Container) unmountHostMounts() error {
 	for _, m := range c.mounts {
 		if m.HostPath != "" {
+			logger := c.Logger().WithField("host-path", m.HostPath)
 			if err := syscall.Unmount(m.HostPath, 0); err != nil {
-				c.Logger().WithFields(logrus.Fields{
-					"host-path": m.HostPath,
-					"error":     err,
-				}).Warn("Could not umount")
+				logger.WithError(err).Warn("Could not umount")
+				return err
```

Comment:
Not a change by your patch, but directly related - it seems if we fail one unmount then we quit the loop. Would it maybe be better for us to continue to try and unmount the remaining mounts in `c.mounts`, and clean up as much as we can (even though we know we will ultimately fail this function)?
Same for the following `RemoveAll` as well.

Reply:
@grahamwhaley  great catch and suggestions~

When I coded these part, it also took me a lot of time to think whether `continue to next` is better than `break from here`.After thinking and discussion with teammates,  and some "sad" experience from our production environment, I think `umount` failure is a very bad situation maybe caused by `mount point leak` and other bad thing... So we need to break right here and raise a warning to upper platform like `k8s`. 

In other word, If a remaining has already happened, maybe it better to alarm the `SRE` department as quick as possible to get involved in this problem.

What do you think about this, glad to hear your suggestion in this situation~~~ ^_^

Reply:
Hi @jshachm Yeah, it is a tricky call if we quit or carry on. I can see your point that 'fail fast' and try not to make things worse is perfectly valid. Maybe that is the safer way which will:
- possibly cause 'less damage' if things are really broken
- give the admins more of a chance to diagnose the problem

So, I'm happy to leave as is. If we do, can we add a big comment saying why we quit on the first failure ;-)
Probably best if we get some other input from maybe @egernst and @gnawux maybe, and others if anybody has a strong opinion on this?